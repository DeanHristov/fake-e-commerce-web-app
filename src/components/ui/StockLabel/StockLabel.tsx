import { FC } from 'react';

export interface IStockLabelProps {
  stock: number;
}

const StockLabel: FC<IStockLabelProps> = ({ stock }) => {
  return (
    <div className="flex items-center space-x-1 mt-5">
      <div
        className={`w-1.5 h-1.5 ${
          stock < 10 ? 'bg-red-500 animate-ping' : 'bg-hover'
        } rounded-full`}
      ></div>
      <span className="text-xs text-black leading-4">{`${stock}+ pcs. in stock`}</span>
    </div>
  );
};

export default StockLabel;
