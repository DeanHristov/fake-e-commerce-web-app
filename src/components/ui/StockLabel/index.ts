import StockLabel, { IStockLabelProps } from './StockLabel';

export type { IStockLabelProps };
export default StockLabel;
