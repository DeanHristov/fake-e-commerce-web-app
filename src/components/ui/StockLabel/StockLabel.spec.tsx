import { render } from '@testing-library/react';
import StockLabel from './';

describe('UI/Component <StockLabel {...} />', () => {
  it('Should be able to render the default state', () => {
    const { container } = render(<StockLabel stock={20} />);

    expect(container).toMatchSnapshot();
  });

  it('Should be able to render a "warning" state', () => {
    const { container } = render(<StockLabel stock={5} />);

    expect(container).toMatchSnapshot();
  });
});
