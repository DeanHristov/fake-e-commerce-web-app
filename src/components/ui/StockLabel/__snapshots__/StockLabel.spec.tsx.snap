// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Component <StockLabel {...} /> Should be able to render a "warning" state 1`] = `
<div>
  <div
    class="flex items-center space-x-1 mt-5"
  >
    <div
      class="w-1.5 h-1.5 bg-red-500 animate-ping rounded-full"
    />
    <span
      class="text-xs text-black leading-4"
    >
      5+ pcs. in stock
    </span>
  </div>
</div>
`;

exports[`UI/Component <StockLabel {...} /> Should be able to render the default state 1`] = `
<div>
  <div
    class="flex items-center space-x-1 mt-5"
  >
    <div
      class="w-1.5 h-1.5 bg-hover rounded-full"
    />
    <span
      class="text-xs text-black leading-4"
    >
      20+ pcs. in stock
    </span>
  </div>
</div>
`;
